provider/execution/execution.go

```diff
@@ -178,45 +165,37 @@ func (e TableExecutor) cleanupStaleData(ctx context.Context, client schema.Clien
 		filters = append(filters, e.Table.DeleteFilter(client, parent)...)
 	}
 	if err := e.Db.RemoveStaleData(ctx, e.Table, e.executionStart, filters); err != nil {
-		e.Logger.Warn("failed to clean table stale data", "last_update", e.executionStart, "err", err)
-		return err
+		e.Logger.Error("failed to clean table stale data", "last_update", e.executionStart, "err", err)
+		return
 	}
 	e.Logger.Debug("cleaned table stale data successfully", "last_update", e.executionStart)
-	return nil
+	return
 }
 
 // callTableResolve does the actual resolving of the table calling the root table's resolver and for each returned resource resolves its columns and relations.
-func (e TableExecutor) callTableResolve(ctx context.Context, client schema.ClientMeta, parent *schema.Resource) (uint64, diag.Diagnostics) {
-	clock := stats.NewClockWithObserve("callTableResolve", segmentStats.Tag{Name: "client_id", Value: identifyClient(client)}, segmentStats.Tag{Name: "table", Value: e.Table.Name})
-	defer clock.Stop()
```

Comment:
why did u remove the stats? this is so we can see when we have hanging resolvers that are still running